v2/pkg/catalog/loader/filter/tag_filter.go

```diff
@@ -25,94 +28,109 @@ var ErrExcluded = errors.New("the template was excluded")
 // matchAllows section.
 //
 // It returns true if the tag is specified, or false.
-func (t *TagFilter) Match(tag, author, severity string) (bool, error) {
-	matchedAny := false
-	if len(t.allowedTags) > 0 {
-		_, ok := t.allowedTags[tag]
-		if !ok {
-			return false, nil
+func (tagFilter *TagFilter) Match(templateTags, templateAuthors []string, templateSeverity severity.Severity) (bool, error) {
+	for _, templateTag := range templateTags {
+		_, blocked := tagFilter.block[templateTag]
+		_, allowed := tagFilter.matchAllows[templateTag]
+
+		if blocked && !allowed { // the whitelist has precedence over the blacklist
+			return false, ErrExcluded
 		}
-		matchedAny = true
 	}
-	_, ok := t.block[tag]
-	if ok {
-		if _, allowOk := t.matchAllows[tag]; allowOk {
-			return true, nil
-		}
-		return false, ErrExcluded
+
+	if !isTagMatch(templateTags, tagFilter) {
+		return false, nil
 	}
-	if len(t.authors) > 0 {
-		_, ok = t.authors[author]
-		if !ok {
-			return false, nil
-		}
-		matchedAny = true
+
+	if !isAuthorMatch(templateAuthors, tagFilter) {
+		return false, nil
 	}
-	if len(t.severities) > 0 {
-		_, ok = t.severities[severity]
-		if !ok {
+
+	if utils.IsNotEmpty(tagFilter.severities) {
+		if _, ok := tagFilter.severities[templateSeverity]; !ok {
 			return false, nil
 		}
-		matchedAny = true
 	}
-	if len(t.allowedTags) == 0 && len(t.authors) == 0 && len(t.severities) == 0 {
-		return true, nil
-	}
-	return matchedAny, nil
+
+	return true, nil
 }
 
-// MatchWithAllowedTags takes an addition list of allowed tags
-// and returns true if the match was successful.
-func (t *TagFilter) MatchWithAllowedTags(allowed []string, tag, author, severity string) (bool, error) {
-	matchedAny := false
+func isAuthorMatch(templateAuthors []string, tagFilter *TagFilter) bool {
+	if utils.IsEmpty(tagFilter.authors) {
+		return true
+	}
 
-	allowedMap := make(map[string]struct{})
-	for _, tag := range allowed {
-		for _, val := range splitCommaTrim(tag) {
-			if _, ok := allowedMap[val]; !ok {
-				allowedMap[val] = struct{}{}
+	for _, templateAuthor := range templateAuthors {
+		if _, ok := tagFilter.authors[templateAuthor]; ok {
+			return true
+		}
+	}
+
+	return false
+}
+
+//nolint:indent-error-flow,revive // keeping conditions together
```

Comment:
I would agree with the `revive` linter

Reply:
On a second thought, the approach I've taken only makes sense if the single return point philosophy is followed, which is not the case here.